packages/plugin-typeorm/src/typeorm.ts

```diff
@@ -0,0 +1,149 @@
+                'delete',
+                'softDelete',
+                'restore',
+                'count',
+                'find',
+                'findAndCount',
+                'findByIds',
+                'findOne',
+                'increment',
+                'decrement',
+            ];
+
+            functionsUsingEntityPersistExecutor.forEach((op) =>
+                shimmer.wrap(
+                    connection.manager,
+                    op as keyof typeorm.EntityManager,
+                    thisPlugin._getEntityManagerFunctionPatch(op).bind(thisPlugin)
+                )
+            );
+
+            functionsUsingQueryBuilder.forEach((op) =>
+                shimmer.wrap(
+                    connection.manager,
+                    op as keyof typeorm.EntityManager,
+                    thisPlugin._getEntityManagerFunctionPatch(op).bind(thisPlugin)
+                )
+            );
+
+            return connection;
+        };
+    }
+
+    private _getEntityManagerFunctionPatch(opName: string) {
+        const thisPlugin = this;
+        thisPlugin._logger.debug(`TypeormPlugin: patched EntityManager ${opName} prototype`);
+        return function (original: Function) {
+            return async function (...args: any[]) {
+                const connectionOptions = this?.connection?.options ?? {}
+                const attributes = {
+                    [DatabaseAttribute.DB_SYSTEM]: connectionOptions.type,
+                    [DatabaseAttribute.DB_USER]: connectionOptions.username,
+                    // [GeneralAttribute.NET_PEER_IP]: '?',
+                    [GeneralAttribute.NET_PEER_NAME]: connectionOptions.host,
+                    [GeneralAttribute.NET_PEER_PORT]: connectionOptions.port,
+                    // [GeneralAttribute.NET_TRANSPORT]: '?',
+                    [DatabaseAttribute.DB_NAME]: connectionOptions.database,
+                    [DatabaseAttribute.DB_OPERATION]: opName,
+                    [DatabaseAttribute.DB_STATEMENT]: JSON.stringify(buildStatement(original, args)),
+                };
+
+                Object.entries(attributes).forEach(([key, value]) => {
+                    if (value === undefined) delete attributes[key];
+                });
+
+                const newSpan: Span = thisPlugin._tracer.startSpan(`TypeORM ${opName}`, {
+                    kind: SpanKind.CLIENT,
+                    attributes,
+                });
+
+                const response: Promise<any> = thisPlugin._tracer.withSpan(newSpan, () =>
+                    original.apply(this, arguments)
+                );
+                try {
+                    const resolved = await response;
+                    if (thisPlugin._config?.responseHook) {
+                        safeExecute([], () => this._config.responseHook(newSpan, resolved), false);
+                    }
+                    newSpan.end();
+                    return resolved;
+                } catch (err) {
```

Comment:
We can do finally and in it newSpan.end()

Reply:
done